### DISCUSSION TITLE
AMD testing

### DISCUSSION THREAD

**Discussion by Lurkki14** [Lurkki14/tuxclocker#60] [Announcements]
Hello

I have added some AMD functionality to the master branch, namely getting the GPU name and temperature.
If you have an AMD card, testing would be appreciated so I know what works and doesn't.

You can post your results here or in the chat room.

**Comment by Nekotje** [comment]
"AMD Ryzen 7 3700X 8-Core Processor"
"Frequencies"
"Core 0"
"Core 1"
"Core 2"
"Core 3"
"Core 4"
"Core 5"
"Core 6"
"Core 7"
"Temperatures"
"Utilizations"
"Core 0"
"Core 1"
"Core 2"
"Core 3"
"Core 4"
"Core 5"
"Core 6"
"Core 7"
"Power Saving Tendencies"
"Governor"
"Governor"
"Core 0"
"Core 1"
"Core 2"
"Core 3"
"Core 4"
"Core 5"
"Core 6"
"Core 7"
"Minimum Frequencies"
"Core 0"
"Core 1"
"Core 2"
"Core 3"
"Core 4"
"Core 5"
"Core 6"
"Core 7"
"Maximum Frequencies"
"Core 0"
"Core 1"
"Core 2"
"Core 3"
"Core 4"
"Core 5"
"Core 6"
"Core 7"
amdgpu_device_initialize: amdgpu_get_auth (1) failed (-1)
/nix/store/3fb92f7hb90llj9dwydx6l8ds4s6v9ry-libdrm-2.4.115/share/libdrm/amdgpu.ids: No such file or directory
"/4e18967c55baab1033250c9f8b0016b1"
"/4e18967c55baab1033250c9f8b0016b1/76b13d2c231dfde5fa4be16ce9503834"
"AMD Radeon Graphics"
"Temperature"
Couldn't register object at path "/4e18967c55baab1033250c9f8b0016b1" QDBusError("", "")
"/4e18967c55baab1033250c9f8b0016b1"
Couldn't register object at path "/4e18967c55baab1033250c9f8b0016b1/76b13d2c231dfde5fa4be16ce9503834" QDBusError("", "")
"/4e18967c55baab1033250c9f8b0016b1/76b13d2c231dfde5fa4be16ce9503834"
"AMD Radeon Graphics"
"Temperature"
Gtk-Message: 09:51:06.435: Failed to load module "canberra-gtk-module"
                                                                      Gtk-Message: 09:51:06.436: Failed to load module "canberra-gtk-module"
                                                            qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
                                                                     qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)

--
it only see the temperature. if i check it against corectl its the same.

**Reply by Lurkki14** [reply]
Yes, only temperature fetching is implemented at the moment. Thanks for checking, now I know that the logic to identify AMD GPUs is correct.

**Comment by Lurkki14** [comment]
Added AMD fan controls and reading now. Let me know how it works.

**Comment by Lurkki14** [comment]
Power limit setting and power usage reading added.

**Comment by tujhen** [comment]
Don't know at all how to change fan speed xD

I change Fan Mode to Manual, change Fan Speed to 24%, apply change and.. seems nothing. PPfeaturemask enabled.

UPD. (Strange, but works when turned back to automatic). So seems little mistake in calculations, when Fan Speed enabled for 27%, GPU fan usage is 20% for example

UPD2. Example 2, when set to 50%, GPU use it for 57%

**Comment by Lurkki14** [comment]
Added voltage-frequency curve editing for the GPUs that have it. The GPUs that use offsets aren't included yet.

**Comment by Lurkki14** [comment]
Attempted to add RX-7000 series fan speed setting with https://github.com/Lurkki14/tuxclocker/commit/4f96ee16288fcb103fbf6a364dddf442a94b1ca1

Users of other cards should try this as well, since it may contain a regression.
